src/modules/esl-carousel/renderers/esl-carousel.css-only.renderer.ts

```diff
@@ -24,119 +26,102 @@ export class ESLCSSCarouselRenderer extends ESLCarouselRenderer {
    * Clear animation.
    */
   public override onUnbind(): void {
-    // TODO: check transformation
-    this.$carousel.toggleAttribute('animate', false);
-    this.$carousel.toggleAttribute('direction', false);
-    this.$carousel.$slides.forEach((slide) => {
-      slide.classList.remove('next');
-      slide.classList.remove('prev');
-    });
+    this.$carousel.toggleAttribute('animating', false);
   }
 
   /** Pre-processing animation action. */
   public async onBeforeAnimate(index: number, direction: ESLCarouselDirection): Promise<void> {
-    if (this.$carousel.hasAttribute('animate')) return Promise.reject();
+    if (this.$carousel.hasAttribute('animating')) throw new Error('[ESL] Carousel: already animating');
 
-    const {$activeSlide} = this.$carousel;
-    if (!$activeSlide) return; // TODO: error
-
-    let $nextSlide = $activeSlide.$nextCyclic;
-    let $prevSlide = $activeSlide.$prevCyclic;
-
-    if (direction === 'prev') $prevSlide = this.$carousel.$slides[index];
-    if (direction === 'next') $nextSlide = this.$carousel.$slides[index];
-
-    $prevSlide.classList.add('prev');
+    const $nextSlide = this.$carousel.$slides[index];
     $nextSlide.classList.add('next');
 
-    this.$carousel.setAttribute('direction', direction);
+    const dir = direction === 'next' ? 'right' : 'left';
+    $nextSlide.classList.add(dir);
+
     return promisifyNextRender();
   }
 
   /** Processes animation. */
-  public async onAnimate(nextIndex: number, direction: ESLCarouselDirection): Promise<void> {
-    this.$carousel.toggleAttribute('animate', true);
+  public async onAnimate(index: number, direction: ESLCarouselDirection): Promise<void> {
+    this.$carousel.toggleAttribute('animating', true);
+
+    const {$activeSlide} = this.$carousel;
+    if (!$activeSlide) throw new Error('[ESL] Carousel: not have active slide');
+
+    const to = direction === 'next' ? 'forward' : 'backward';
+
+    const $nextSlide = this.$carousel.$slides[index];
+    $nextSlide.classList.add('next');
+
+    const dir = direction === 'next' ? 'right' : 'left';
+    $nextSlide.classList.add(dir);
+
+    $activeSlide.classList.add(to);
+    $nextSlide.classList.add(to);
 
-    // TODO: !
     return promisifyEvent(this.$carousel.$slidesArea, 'transitionend').catch(resolvePromise);
```

Comment:
1) TODO: we need to leave some more options to catch this animation. E.g. make user able to use animation instead of transition.
2) It looks like it works (or may works incorrectly) from targeting perspective. We need to observe for transition of *specific properties* and for *slide itself*

3) We need to create good promisifyTransition/promisifyAnimation option + Note we have a global bug with non considering `transitioncancel` and the same problem in default renderer


In terms of current PR it is only required to have a noticeable TODO with mentioned undone factors